sim/midas/src/main/cc/simif_vitis.cc

```diff
@@ -0,0 +1,103 @@
+#include "simif_vitis.h"
+#include <cassert>
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+constexpr size_t u250_dram_channel_size_bytes = 16ULL * 1024 * 1024 * 1024;
+
+simif_vitis_t::simif_vitis_t(int argc, char** argv) {
+    device_index = -1;
+    binary_file = "";
+
+    // TODO: Properly read out arguments
+    std::vector<std::string> args(argv + 1, argv + argc);
+    for (auto &arg: args) {
+        if (arg.find("+device_index=") == 0) {
+            device_index = atoi((arg.c_str()) + 14); // 0
+        }
+        if (arg.find("+binary_file=") == 0) {
+            binary_file = arg.substr(13, std::string::npos);//"kernel.xclbin"; //atoi((arg.c_str()) + 8);
+        }
+    }
+
+    if (device_index == -1) {
+        fprintf(stderr, "Device ID not specified. Assuming Device 0.\n");
+        device_index = 0;
+    }
+
+    if (binary_file == "") {
+        fprintf(stderr, "No binary file specified.\n");
+        exit(1);
+    }
+
+    fprintf(stdout, "DEBUG: DevIdx:%d XCLBin:%s\n", device_index, binary_file.c_str());
+
+    // Open the FPGA device
+    device_handle = xrt::device(device_index);
+
+    // Load the XCLBIN (get handle, then load)
+    uuid = device_handle.load_xclbin(binary_file);
+
+    // Open Kernel
+    kernel_handle = xrt::ip(device_handle, uuid, "firesim");
```

Comment:
I'm not sure how you change this to be a two-stage load. @timsnyder-siv do you happen to know (since you looked at `xbutil` in more depth than me).

Reply:
For this code in the driver, you just need to change line 42 to `uuid = device_handle.get_xclbin_uuid()` to get the currently loaded xclbin uuid.  I think we should print that out similar to how we print the AGFI for f1.  The docs don't say what happens if the device doesn't have a xclbin loaded.

In infrasetup in the other PR, you will need to use `xbutil program -d {BDF} {xclbin}` like I mentioned to do the programming before starting the driver.